servo/connectors/kubernetes.py

```diff
@@ -3125,29 +3140,34 @@ async def apply(self, adjustments: List[servo.Adjustment]) -> None:
                 f"waiting for adjustments to take effect on {len(self.optimizations)} optimizations"
             )
             try:
-                results = await asyncio.wait_for(
-                    asyncio.gather(
-                        *list(map(lambda a: a.apply(), self.optimizations)),
-                        return_exceptions=True,
-                    ),
-                    timeout=timeout.total_seconds() + 60, # allow sub-optimization timeouts to expire first
+                gather_apply = asyncio.gather(
+                    *list(map(lambda a: a.apply(), self.optimizations)),
+                    return_exceptions=True,
                 )
-
-                for result in results:
-                    if isinstance(result, Exception):
-                        for optimization in self.optimizations:
-                            if await optimization.handle_error(result):
-                                # Stop error propagation once it has been handled
-                                break
+                results = await asyncio.wait_for(gather_apply, timeout=timeout.total_seconds() + 60) # allow sub-optimization timeouts to expire first
```

Comment:
Curious about the need for a static value for additional time.  If this is a parameter, should we set it to a default and let the end user break things if they set it too short?

Reply:
The rationale for the extra 1 minute buffer in this case is that this particular timeout is applied to all configured optimizations (of which there can be multiple) aggregated under the KubernetesOptimizations class. However each sub-optimization ([DeploymentOptimization](https://github.com/opsani/servox/blob/3c8f47cc9f630e85f0999b2f4709fe47178dda1d/servo/connectors/kubernetes.py#L2453), [CanaryOptimization](https://github.com/opsani/servox/blob/3c8f47cc9f630e85f0999b2f4709fe47178dda1d/servo/connectors/kubernetes.py#L2773)) also has the configured timeout applied to its own operations.

If KubernetesOptimizations' timeout were the exact same as the sub-timeouts, it would be the first to expire and obfuscate information about which sub-optimization was the one that actually caused the timeout